model/headers/headers.go

```diff
@@ -306,8 +306,7 @@ type EncryptedHeaderPacket interface {
 // NewEncryptedHeaderPacket method constructs EncryptedHeaderPacket from io.Reader and supplied private key.
 func NewEncryptedHeaderPacket(encryptedPacketPayload []byte, headerEncryptionMethod HeaderEncryptionMethod, readerPrivateKey [chacha20poly1305.KeySize]byte) (*EncryptedHeaderPacket, error) {
 	var encryptedHeaderPacket EncryptedHeaderPacket
-	switch headerEncryptionMethod {
-	case X25519ChaCha20IETFPoly1305:
+	if headerEncryptionMethod == X25519ChaCha20IETFPoly1305 {
```

Comment:
This seems un necessary since we only support one method at the moment

Reply:
added comments in case we forget